### PR TITLE
Newick service

### PR DESCRIPTION
**Pull request by d-callan**
we have as a higher level objective replacing the cluster graph on ortho group record pages w a protein tree. the workflow produces newick files w the data for the tree, and we need a way to get these files from web services to tidytree. that is this pr, hopefully.

### CODE REVIEW

**Review by ryanrdoherty — COMMENTED** [review]
Looks ok so far.  I would put the path variable into the method signature now, even if you're not going to use it yet.  The catch should throw an exception, not just print stack trace.  We typically don't use System.out/err because that goes to catalina.out.  You want log4j for diagnostic output.  But looks like both places it's just a placeholder for now.  I would say: "proceed!"

File: Service/src/main/java/org/orthomcl/service/services/NewickProteinTreeService.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public Response getNewickProteinTree(@PathParam("orthoGroupId") String orthoGroupId) throws WdkModelException {
+        String projectId = getWdkModel().getProjectId();
+        String buildNumber = getWdkModel().getBuildNumber();
+        String webservicesDir = getWdkModel().getProperties().get("WEBSERVICEMIRROR");
+        // TODO need to add orthoGroupId to this path, update to reflect actual webservice mirror directory structure
+        // this should be considered a placeholder really
+        String newickPath = webservicesDir + "/" + projectId + "/" + "build-" + buildNumber + "/data/newick";
+
+        try (BufferedReader br = new BufferedReader(new FileReader(newickPath))) {
+            String newick = "";
```
Comment (ryanrdoherty) [inline Service/src/main/java/org/orthomcl/service/services/NewickProteinTreeService.java]: Use a StringBuilder here.  Using `+=` creates a new String object every time and copies the array contents into the new string before adding the contents of the right operand.  Once you're done, call `toString()` to create the String.

File: Service/src/main/java/org/orthomcl/service/services/NewickProteinTreeService.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            while ((String line = br.readLine()) != null) {
+                // not even sure we should have more than one line?
+                // need to check files when we have them
+                newick += line;
+            }
+            LOG.debug("Newick: " + newick);
+        }
+        catch (IOException e) {
```
Comment (ryanrdoherty) [inline Service/src/main/java/org/orthomcl/service/services/NewickProteinTreeService.java]: Once you start using the path param, may want to catch FileNotFoundException and throw a NotFoundException (JAX-RS exception which will trigger a 404).

File: Service/src/main/java/org/orthomcl/service/services/NewickProteinTreeService.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    private static final Logger LOG = Logger.getLogger(NewickProteinTreeService.class);
+
+    @GET
+    @Path ("/{orthoGroupId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getNewickProteinTree(@PathParam("orthoGroupId") String orthoGroupId) throws WdkModelException {
+        String projectId = getWdkModel().getProjectId();
+        String buildNumber = getWdkModel().getBuildNumber();
+        String webservicesDir = getWdkModel().getProperties().get("WEBSERVICEMIRROR");
+        // TODO need to add orthoGroupId to this path, update to reflect actual webservice mirror directory structure
```
Comment (ryanrdoherty) [inline Service/src/main/java/org/orthomcl/service/services/NewickProteinTreeService.java]: Once you start using the path param, will have to do some input validation to make sure someone can't input relative paths.

**Review by ryanrdoherty — APPROVED** [review]

File: Service/src/main/java/org/orthomcl/service/services/NewickProteinTreeService.java (hunk 3 above)
Comment (d-callan) [reply Service/src/main/java/org/orthomcl/service/services/NewickProteinTreeService.java]: Do you have an ex how you'd recommend to do this? Are we talking about something very simple like checking for `/` and `..` or ....?

**Comment by d-callan** [comment]
tested and confirmed working w the demo files provided by @rdemko2332

**Pull request merged**